### ISSUE TITLE
Bus error attempting to create ragged array (migrated from Trac #1078)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#2631]
Original ticket http://projects.scipy.org/numpy/ticket/1078
Reported 2009-04-06 by trac user anand, assigned to unknown.

This has been observed as recently as 1.4.0.dev6728:
    In [1]: import numpy as np

```
In [2]: a=np.ones(2)

In [3]: b=np.array(3)

In [4]: np.array((a,b))
Segmentation fault
```

**Comment by thouis** [comment]
Comment in Trac by atmention:nilswagner01, 2009-04-06

I can reproduce the problem. Here is a backtrace:
    Program received signal SIGSEGV, Segmentation fault.
    0x00007ffff6c0ed74 in discover_dimensions (s=0x89ff90, nd=1, d=0x7fffffffd058, check_it=1)
        at numpy/core/src/arrayobject.c:7615
    7615                d[i] = PyArray_DIM(s,i);
    (gdb) bt
    #0  0x00007ffff6c0ed74 in discover_dimensions (s=0x89ff90, nd=1, d=0x7fffffffd058, check_it=1)
        at numpy/core/src/arrayobject.c:7615
    #1  0x00007ffff6c0ee3b in discover_dimensions (s=0x7ffff4651878, nd=<value optimized out>, d=<value optimized out>,
        check_it=1) at numpy/core/src/arrayobject.c:7632
    #2  0x00007ffff6c2637c in Array_FromSequence (s=0x7ffff4651878, typecode=0x7ffff6e6f660, fortran_=0,
        min_depth=<value optimized out>, max_depth=0) at numpy/core/src/arrayobject.c:8208
    #3  0x00007ffff6c268b3 in PyArray_FromAny (op=0x7ffff4651878, newtype=0x7ffff6e6f660, min_depth=0, max_depth=0,
        flags=<value optimized out>, context=<value optimized out>) at numpy/core/src/arrayobject.c:9401
    #4  0x00007ffff6c35e2b in PyArray_CheckFromAny (op=0x7ffff4651878, descr=0x0, min_depth=0, max_depth=0,
        requires=112, context=0x0) at numpy/core/src/arrayobject.c:9559
    #5  0x00007ffff6c37b0e in _array_fromobject (__NPY_UNUSED_TAGGEDignored=<value optimized out>,
        args=<value optimized out>, kws=<value optimized out>) at numpy/core/src/multiarraymodule.c:6329
    #6  0x00007ffff7b15f28 in PyEval_EvalFrameEx () from /usr/lib64/libpython2.6.so.1.0
    #7  0x00007ffff7b1b6c0 in PyEval_EvalCodeEx () from /usr/lib64/libpython2.6.so.1.0
    #8  0x00007ffff7b13822 in PyEval_EvalCode () from /usr/lib64/libpython2.6.so.1.0
    #9  0x00007ffff7b34b13 in ?? () from /usr/lib64/libpython2.6.so.1.0
    #10 0x00007ffff7b34bca in PyRun_FileExFlags () from /usr/lib64/libpython2.6.so.1.0
    #11 0x00007ffff7b3550d in PyRun_SimpleFileExFlags () from /usr/lib64/libpython2.6.so.1.0
    #12 0x00007ffff7b40dba in Py_Main () from /usr/lib64/libpython2.6.so.1.0
    #13 0x00007ffff6e94586 in __libc_start_main () from /lib64/libc.so.6
    #14 0x00000000004006e9 in _start ()

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2009-04-10

I think this is related to ticket #1081. Again, reversing the order of a,b leads to a different error:

```
In [11]: array((b,a))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)

/home/charris/<ipython console> in <module>()

ValueError: setting an array element with a sequence.
```

**Comment by thouis** [comment]
Comment in Trac by trac user jary, 2009-07-09

I think I came across same problem running

```
import numpy as np
t = ((1,), np.array(1))
np.array(t)
```

Backtrace is:

```
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread -1211758912 (LWP 19759)]
0xb78c51c6 in discover_dimensions (s=0x84faa50, nd=1, d=0xbfa70f8c, check_it=1)
    at numpy/core/src/arrayobject.c:7586
7586    numpy/core/src/arrayobject.c: No such file or directory.
    in numpy/core/src/arrayobject.c
(gdb) bt
#0  0xb78c51c6 in discover_dimensions (s=0x84faa50, nd=1, d=0xbfa70f8c, check_it=1)
    at numpy/core/src/arrayobject.c:7586
#1  0xb78c5274 in discover_dimensions (s=0xb7663d6c, nd=2, d=0xbfa70f88, check_it=1)
    at numpy/core/src/arrayobject.c:7603
#2  0xb78de345 in PyArray_FromAny (op=0xb7663d6c, newtype=0xb7917580, min_depth=0, max_depth=2, flags=112, 
    context=0x0) at numpy/core/src/arrayobject.c:8179
#3  0xb78ee466 in PyArray_CheckFromAny (op=0xb7663d6c, descr=0x0, min_depth=0, max_depth=0, requires=112, 
    context=0x0) at numpy/core/src/arrayobject.c:9530
#4  0xb78f0141 in _array_fromobject (__NPY_UNUSED_TAGGEDignored=0x0, args=0xb7bf6dec, kws=0x0)
    at numpy/core/src/multiarraymodule.c:6329
#5  0xb7e5c88a in PyCFunction_Call () from /usr/lib/libpython2.5.so.1.0
#6  0xb7ea521d in PyEval_EvalFrameEx () from /usr/lib/libpython2.5.so.1.0
#7  0xb7ea5d80 in PyEval_EvalCodeEx () from /usr/lib/libpython2.5.so.1.0
#8  0xb7ea5df5 in PyEval_EvalCode () from /usr/lib/libpython2.5.so.1.0
#9  0xb7eb5647 in PyImport_ExecCodeModuleEx () from /usr/lib/libpython2.5.so.1.0
#10 0xb7eb5b4a in ?? () from /usr/lib/libpython2.5.so.1.0
#11 0xbfa734f7 in ?? ()
#12 0xb7bea458 in ?? ()
#13 0xbfa7141b in ?? ()
#14 0x79702e38 in ?? ()
#15 0x08040063 in ?? ()
#16 0xb7bea068 in ?? ()
#17 0xffffffff in ?? ()
---Type <return> to continue, or q <return> to quit---
```

Code in ticket notice message (above) also segfaults for me.

**Comment by thouis** [comment]
Comment in Trac by atmention:cournape, 2009-11-27

Fixed in r7787 for 1.4.x branch and r7785 in the trunk
